include/envoy/network/connection.h

```diff
@@ -116,6 +116,13 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
   virtual const std::string& remoteAddress() PURE;
 
   /**
+   * @return The address the remote client is trying to connect to.
+   * It can be different from the proxy address if the downstream connection
+   * has been redirected or the proxy is operating in transparent mode.
+   */
+  virtual const std::string destinationAddress() PURE;
```

Comment:
yes, that makes sense. The only issue is that it becomes inaccurate when the connection is intercepted. It's probably ok since the comment above mentions that explicitly.
Anyway, I will also try to think of a better name.